app/scripts/metamask-controller.js

```diff
@@ -177,7 +177,7 @@ module.exports = class MetamaskController extends EventEmitter {
       blockTracker: this.blockTracker,
       getGasPrice: this.getGasPrice.bind(this),
     })
-    this.txController.on('newUnapprovedTx', opts.showUnapprovedTx.bind(opts))
+    this.txController.on('newUnapprovedTx', () => opts.showUnapprovedTx())
```

Comment:
This one's just for readability, right? There isn't some issue with bind, is there?

Reply:
So there is a lot of controversy around bind vs arrow functions with arrow functions winning the performance battle since they only bind once but the reason I switched this one was because ‘opts’ was being bound. I am not sure whether it should be like that but when I analyzed the heap, it looked like it was cloning the opts every time it bound

Reply:
Fair enough, looks like arrow functions make it harder to mess up as usual! Thanks for the improvement. Getting the tests passing, they can be a little non-deterministic :(

Should be able to merge soon!